arkouda/segarray.py

```diff
@@ -1070,3 +1071,29 @@ def attach(cls, name, segment_suffix='_segments', value_suffix='_values', length
         # TODO - add grouping attaching grouping=ak.GroupBy.attach(name+grouping_suffix)
         return cls(attach_pdarray(name+segment_suffix), attach_pdarray(name+value_suffix),
                    lengths=attach_pdarray(name+length_suffix))
+
+    def is_registered(self) -> bool:
+        """
+        Checks if all components of the SegArray object are registered
+
+        Returns
+        -------
+        bool
+            True if all components are registered, false if not
+
+        See Also
+        --------
+        register, unregister, attach
+        """
+
+        # SegArray contains 3 parts - segments, values, and lengths
+        regParts = sum([self.segments.is_registered(), self.values.is_registered(), self.lengths.is_registered()])
+
+        if regParts == 3:
+            return True
+        elif regParts == 0:
+            return False
+        else:
+            warn(f"SegArray expected 3 components to be registered, but only located {regParts}")
+            return False
```

Comment:
It's personal preference, but I really like using `any` and `all`, I feel it's more descriptive since reading the `if` tells you exactly what the problem is

```python
parts_registered = [self.segments.is_registered(), self.values.is_registered(), self.lengths.is_registered()]

if any(parts_registered) and not all(parts_registered):
  warn(f"SegArray expected {len(parts_registered)} components to be registered, but only located {sum(parts_registered)}")
return all(parts_registered)    
```

this has the added benefit of requiring fewer lines be changed if we add more registerable components (such as `self.grouping`)